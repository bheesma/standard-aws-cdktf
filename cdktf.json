{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "679d25ee-31c0-4b01-9e78-f9e68dcaa0e4",
  "sendCrashReports": "false",
  "terraformProviders": [
    "hashicorp/aws@~> 4.60.0"
  ],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}