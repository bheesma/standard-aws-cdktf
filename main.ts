import { Construct } from "constructs";
import { App, TerraformStack } from "cdktf";
import { SqsQueue } from "@cdktf/provider-aws/lib/sqs-queue";
import  { AwsProvider } from "@cdktf/provider-aws/lib/provider";

class MyStack extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // define resources here

    new AwsProvider(this,'aws',{
      region: "ap-southeast-2"
    });

    const queueName = 'MyQueue2';
    new SqsQueue(this, queueName, {
      name: queueName
    });



  }
}

const app = new App();
new MyStack(app, "app");
app.synth();
