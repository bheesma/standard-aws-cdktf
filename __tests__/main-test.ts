// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import "cdktf/lib/testing/adapters/jest"; // Load types for expect matchers
import { App, Testing } from "cdktf";
import { MyStack } from "../main"
import { SqsQueue } from "@cdktf/provider-aws/lib/sqs-queue";

describe("My CDKTF Application", () => {

  let app: App;
  let stack: MyStack;

  beforeEach(() => {
    app = Testing.app();
    stack = new MyStack(app, 'test');
  });

  // // All Unit tests test the synthesised terraform code, it does not create real-world resources
  describe("Unit testing using assertions", () => {
    it("should contain a resource", () => {

      expect(
        Testing.synth((stack))
      ).toHaveResource(SqsQueue);

      expect(
        Testing.synth((stack))
      ).toHaveResourceWithProperties(SqsQueue, {
        name: "MyQueue2"
      });

    });
  });


  describe("Unit testing using snapshots", () => {
    it("Tests the snapshot", () => {
      const app = Testing.app();
      const stack = new MyStack(app, "test");


      expect(Testing.synth(stack)).toMatchSnapshot();
    })
  });

  describe("Checking validity", () => {
    it("check if the produced terraform configuration is valid", () => {
      const app = Testing.app();
      const stack = new MyStack(app, "test");
  
      // We need to do a full synth to validate the terraform configuration
      expect(Testing.fullSynth(stack)).toBeValidTerraform();
    });

  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new MyStack(app, "test");

    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });

});
