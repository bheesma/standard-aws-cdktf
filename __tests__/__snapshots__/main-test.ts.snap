// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My CDKTF Application Unit testing using snapshots Tests the snapshot 1`] = `
"{
  "provider": {
    "aws": [
      {
        "region": "ap-southeast-2"
      }
    ]
  },
  "resource": {
    "aws_sqs_queue": {
      "MyQueue2": {
        "name": "MyQueue2"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "4.60.0"
      }
    }
  }
}"
`;
